### ISSUE TITLE
Are you using test rating matrix during matrix completion??

### ISSUE THREAD

**Issue by sh0416** [riannevdberg/gc-mc#3]
I analyze your code and I think you use test rating matrix is used while completing test rating matrix.

Is it real?????

I judge following code.

```
test_support = support[np.array(test_u)]
test_support_t = support_t[np.array(test_v)]
```

This part save the things which we need to predict. Right?

Also, when GCMC do encoding,

```
self.layers.append(OrdinalMixtureGCN(input_dim=self.input_dim,
                                                 output_dim=self.hidden[0],
                                                 support=self.support,
                                                 support_t=self.support_t,
                                                 num_support=self.num_support,
                                                 u_features_nonzero=self.u_features_nonzero,
                                                 v_features_nonzero=self.v_features_nonzero,
                                                 sparse_inputs=True,
                                                 act=tf.nn.relu,
                                                 bias=False,
                                                 dropout=self.dropout,
                                                 logging=self.logging,
                                                 share_user_item_weights=True,
                                                 self_connections=False))
```

In this code, GCN use support to encode, which is predicted itself.

If I have mistake, please correct me.

I think we only use training matrix to complete matrix and test matrix is used only for calculating RMSE or some evaluation metric.

**Comment by sh0416** [comment]
My mistake, you use only node information not relational information.
